### PR TITLE
Collect trace_api.{requests,responses,errors} metrics in the background trace sender

### PR DESCRIPTION
**Pull request by iamluc**
### Reviewer checklist
- [ ] Test coverage seems ok.
- [ ] Appropriate labels assigned.

### CODE REVIEW

File: ext/telemetry.c [hunk 1]
```diff
@@ -167,3 +191,45 @@ void ddtrace_telemetry_inc_spans_created(ddtrace_span_data *span) {
...
+    }
+    if (metrics.errors_network) {
+        ddog_sidecar_telemetry_add_span_metric_point_buffer(buffer, DDOG_CHARSLICE_C("trace_api.errors"), (double)metrics.errors_network, DDOG_CHARSLICE_C("type:network"));
+    }
+    if (metrics.errors_status_code) {
+        ddog_sidecar_telemetry_add_span_metric_point_buffer(buffer, DDOG_CHARSLICE_C("trace_api.errors"), (double)metrics.errors_status_code, DDOG_CHARSLICE_C("type:status_code"));
+    }
+
+    ddog_sidecar_telemetry_buffer_flush(&ddtrace_sidecar, ddtrace_sidecar_instance_id, &dd_bgs_queued_id, buffer);
+    ddog_sidecar_telemetry_flush(&ddtrace_sidecar, ddtrace_sidecar_instance_id, &dd_bgs_queued_id);
```
Comment (bwoebi) [inline ext/telemetry.c:234]: Are we doing that for tests? Or does it really need to be instantaneously flushed?

Comment (iamluc) [reply ext/telemetry.c:234]: It's not needed. Without this call, data are flushed every 60 seconds

**Review by pierotibou — COMMENTED** [review]
LGTM funtionnaly speaking, just a comment about retries

File: ext/coms.c [hunk 2]
```diff
@@ -810,74 +810,99 @@ static size_t _dd_curl_writefunc(char *ptr, size_t size, size_t nmemb, void *s)
...
+                smart_str_free_ex(&response, true);
+            }
         }
 
-        _dd_deinit_read_userdata(read_data);
-        _dd_curl_reset_headers(writer);
+        break;
+    }
+
+    // Collect metrics for the last performed HTTP query
```
[the remark this answers is not in the document]
Comment (iamluc) [reply ext/coms.c:876]: I believe it's the same logic that in the sidecar. See https://github.com/DataDog/libdatadog/pull/418#discussion_r1607048897

[the remark this answers is not in the document]
Comment (pierotibou) [reply ext/coms.c:876]: Indeed, forgot about that, thanks for raising. Though, I'm not sure it's the right move to make... Indeed, if we want to be able to compare, we need to make sure this is how things have been implemented in other languages. The thing is, maybe all languages have implemented it differently, so maybe having a simple solution works and we just need to know the differences. Anyway, let's discuss that with everyone.

**Review by bwoebi — APPROVED** [review]
Things seem green now :-)

**Pull request merged**